packages/block-editor/src/components/inserter/menu.js

```diff
@@ -438,12 +469,15 @@ export default compose(
 			}
 		}
 		const destinationRootBlockName = getBlockName( destinationRootClientId );
+		const hiddenBlockTypes = getPreference( 'hiddenBlockTypes' );
+		const numberOfHiddenBlocks = isArray( hiddenBlockTypes ) && hiddenBlockTypes.length;
```

Comment:
This const will return „false” when „hiddenBlockTypes” is not an array. Is there any case where it would have other type?

Reply:
I am not overly familiar with `getPreference` and this is how it was done in the [block manager](https://github.com/WordPress/gutenberg/blob/a491a79ab2ec9c05b296c50e1ce0c5e389e19175/packages/edit-post/src/components/manage-blocks-modal/manager.js#L97-L98), so I did it here as well. In looking into it further though, `hiddenBlockTypes` is defaulted to `[]` and it may not be necessary.

My thought/assumption at the time of adding it was if the block-editor is used in another context in the future, that preference may not exist, and the isArray() would prevent hiddenBlockTypes.length from erroring in that case, but that doesn't look like it is accurate. 

Pinging @jorgefilipecosta as it looks like he added the message to the block manager and may have some insight as to a reason it is needed, otherwise I can remove it here.

Reply:
Hi @brentswisher, normally when we have a list of blocks, boolean false means no blocks and true means all blocks, that the case we followed for allowed blocks.
Andrew original block manager code used as size lodash function https://github.com/WordPress/gutenberg//blob/4698c304a19d81c7f27f8c49e9cf9c4babd7309f/packages/edit-post/src/editor.js#L59
The tests for using hiddenBlockTypes don't account for booleans so I'm not sure if in this case we are following the all/none convention but writing the code I preferred to play on the safe side, also as referred it may avoid a case where the preference because of some reason does not exists.

I guess I should have also just used size and it would cleaner, here I think we can also use size.